api/client/joinservice.go

```diff
@@ -202,6 +207,45 @@ func (c *JoinServiceClient) RegisterUsingTPMMethod(
 	return certs, nil
 }
 
+// RegisterUsingOracleMethod registers the caller using the Oracle join method and
+// returns signed certs to join the cluster. The caller must provide a
+// ChallengeResponseFunc which returns a *proto.RegisterUsingOracleMethodRequest
+// for a given challenge, or an error.
+func (c *JoinServiceClient) RegisterUsingOracleMethod(ctx context.Context, challengeResponse RegisterOracleChallengeResponseFunc) (*proto.Certs, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	oracleJoinClient, err := c.grpcClient.RegisterUsingOracleMethod(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	challengeResp, err := oracleJoinClient.Recv()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	challenge, ok := challengeResp.Response.(*proto.RegisterUsingOracleMethodResponse_Challenge)
+	if !ok {
+		return nil, trace.BadParameter("expected challenge response type %T, got %T", challenge, challengeResp.Response)
+	}
```

Comment:
instead of the type assertion you can do `challengeResp.GetChallenge()` and make sure it's not empty